bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -95,26 +97,35 @@ void ClientsManager::loadInfoFromReservedPages() {
   for (auto const& [clientId, clientIdx] : clientIdToIndex_) {
     const uint32_t firstPageId = clientIdx * reservedPagesPerClient_;
 
-    if (!stateTransfer_->loadReservedPage(resPageOffset() + firstPageId, sizeOfReservedPage_, scratchPage_)) continue;
-
-    ClientReplyMsgHeader* replyHeader = (ClientReplyMsgHeader*)scratchPage_;
-    ConcordAssert(replyHeader->msgType == 0 || replyHeader->msgType == MsgCode::ClientReply);
-    ConcordAssert(replyHeader->currentPrimaryId == 0);
-    ConcordAssert(replyHeader->replyLength >= 0);
-    ConcordAssert(replyHeader->replyLength + sizeof(ClientReplyMsgHeader) <= maxReplySize_);
-
-    // YS TBD: Multiple replies for client batching should be sorted by incoming time
-    auto& repliesInfo = indexToClientInfo_.at(clientIdx).repliesInfo;
-    if (repliesInfo.size() >= maxNumOfReqsPerClient_) deleteOldestReply(clientId);
-    const auto& res = repliesInfo.insert_or_assign(replyHeader->reqSeqNum, MinTime);
-    const bool added = res.second;
-    LOG_INFO(CL_MNGR, "Added/updated reply message" << KVLOG(clientId, replyHeader->reqSeqNum, added));
-
+    // load all replies into the map of replies one by one
+    for (int i = 0; i < maxNumOfReqsPerClient_; i++) {
+      if (!stateTransfer_->loadReservedPage(
```

Comment:
Should not we deserialize replies from reserved pages?

Reply:
Since we are writing and reading reply by reply we didn't actually changed anything.